src/webhooks/webhooks.service.ts

```diff
@@ -67,4 +83,56 @@ export class WebhooksService {
       return error;
     }
   }
+
+  async upsertStory({ action, story_id }: StoryDto) {
+    const {
+      data: { story },
+    } = await Storyblok.get(`cdn/stories/${story_id}`);
+
+    if (!story) {
+      throw new HttpException('STORY NOT FOUND', HttpStatus.NOT_FOUND);
+    }
+
+    const createCourseObject = this.courseRepository.create({
+      name: story.name,
+      slug: story.full_slug,
+      status: action,
+      parent_id: story.parent_id,
+      storyblokId: story_id,
+    });
+
+    try {
+      if (story.content?.component === 'Course') {
+        await this.courseRepository
+          .createQueryBuilder('course')
+          .insert()
+          .into(CourseEntity)
+          .values(createCourseObject)
+          .onConflict(`("storyblokId") DO UPDATE SET "status" = '${action}'`)
+          .execute();
+
+        return createCourseObject;
+      } else if (story.content?.component === 'Session') {
+        const { id } = await this.courseRepository.findOne({ parent_id: story.parent_id });
```

Comment:
To get the course id from the session, you can also use `story.content.course` as `course` is a required field on a session story. i'd trust that relationship over `parent_id` as that's relational rather than where they're placed in the storyblok folders

Reply:
i think that means you can remove the next few lines and skip to
`const sessionObject = this.sessionRepository.create({ courseId: story.content.course });`